src/galaxy/config.py

```diff
@@ -1,13 +1,79 @@
 #!/usr/bin/env python
 # -*- coding: utf-8 -*-
+# Copyright (C) 2021 Humanitarian OpenStreetmap Team
+
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+# Humanitarian OpenStreetmap Team
+# 1100 13th Street NW Suite 800 Washington, D.C. 20005
+# <info@hotosm.org>
 
 from configparser import ConfigParser
+import logging
 
 CONFIG_FILE_PATH = "src/config.txt"
 
 config = ConfigParser()
 config.read(CONFIG_FILE_PATH)
 
+log_level = config.get("API_CONFIG", "log_level",fallback=None) # get log level from config
+
+if log_level is None or log_level.lower() == 'debug':  # default will go to debug
+    level=logging.DEBUG 
+elif log_level.lower() == 'info' :
+    level=logging.INFO
+elif log_level.lower() == 'error' :
+    level= logging.ERROR
+elif log_level.lower() == 'warning':
+    level = logging.WARNING
+else :
+    logging.error("logging config is not supported , Supported fields are : debug,error,warning,info , Logging to default :debug")
+    level = logging.DEBUG
+
+logging.getLogger("fiona").propagate = False  # disable fiona logging
+logging.basicConfig(format='%(asctime)s - %(message)s', level=level)
+logging.getLogger('boto3').propagate = False # disable boto3 logging
+
+logger = logging.getLogger('galaxy')
+
+export_path=config.get('API_CONFIG', 'export_path', fallback=None)
+if export_path is None : 
+    export_path = "exports/"
+if export_path.endswith("/") is False : 
+    export_path=f"""{export_path}/"""
+
+AWS_ACCESS_KEY_ID,AWS_SECRET_ACCESS_KEY , BUCKET_NAME =None , None , None
+#check either to use connection pooling or not 
+use_connection_pooling = config.getboolean("API_CONFIG","use_connection_pooling", fallback=False)
+
+#check either to use s3 raw data exports file uploading or not 
+file_upload_method=config.get("EXPORT_UPLOAD", "FILE_UPLOAD_METHOD",fallback='disk').lower()
+if   file_upload_method== "s3" :
+    use_s3_to_upload=True
+    try :
+        AWS_ACCESS_KEY_ID=config.get("EXPORT_UPLOAD", "AWS_ACCESS_KEY_ID") 
+        AWS_SECRET_ACCESS_KEY=config.get("EXPORT_UPLOAD", "AWS_SECRET_ACCESS_KEY")
+    except :
+        logging.debug("No aws credentials supplied")
```

Comment:
I think we should just assume that credentials are supplied properly. In deployments on AWS, we won't supply AWS credentials. We will simply give the instance a role with the relevant permissions necessary for accessing AWS services. This block will raise an exception merely because credentials are not EXPLICITLY provided - which is limiting for our deployment methods.

Refer: https://boto3.amazonaws.com/v1/documentation/api/latest/guide/credentials.html

Reply:
yeah ! thats right, it will not raise exception its a debug message that says credentials are not supplied , if they are not supplied I will add if clause that it will connect directly -- it will assume aws credentials are setup on the machine already , implementing your previous comment